mgnify_pipelines_toolkit/utils/rename_contig/cli_rename.py

```diff
@@ -0,0 +1,253 @@
+      Creates: output_1kb_contigs.fasta (>1000bp)
+               output_5kb_contigs.fasta (>5000bp)
+               output_100kb_contigs.fasta (>=100000bp)
+
+    \b
+    USING EXISTING MAPPINGS:
+      mpt rename -i new.fasta -o outdir --use-mapping mapping.tsv
+
+    \b
+    SUPPORTED FILE FORMATS:
+      FASTA:   .fasta, .fa, .fna, .fasta.gz, .fa.gz, .fna.gz
+      GFF:     .gff, .gff3, .gff.gz, .gff3.gz
+      GenBank: .gbk, .gb, .genbank
+
+    :param input: Input file(s) (FASTA, GFF, and/or GenBank)
+    :param outdir: Output directory
+    :param prefix: Prefix for new sequence names
+    :param map: Output mapping file (TSV)
+    :param use_mapping: Use existing mapping file instead of generating new names
+    :param from_col: Source column in mapping file
+    :param to_col: Target column in mapping file
+    :param parser: Header parser to use ('base' or 'virify')
+    :param formatter: Metadata formatter to use ('simple', 'generic', or 'virify')
+    :param separate_by_size: Separate output by contig size
+    """
+    # Create output directory if it doesn't exist
+    outdir_path = Path(outdir)
+    outdir_path.mkdir(parents=True, exist_ok=True)
+
+    # Select parser function
+    parser_fn = parse_virify_header if parser == "virify" else parse_header
+
+    # Categorize input files by type
+    fasta_files = []
+    gff_files = []
+    gbk_files = []
+
+    for input_file in input:
+        input_path = Path(input_file)
+        file_ext = "".join(input_path.suffixes).lower()
+
+        is_fasta = any(file_ext.endswith(ext) for ext in SUPPORTED_FASTA_EXTS)
+        is_gff = any(file_ext.endswith(ext) for ext in SUPPORTED_GFF_EXTS)
+        is_gbk = any(file_ext.endswith(ext) for ext in SUPPORTED_GBK_EXTS)
+
+        if is_fasta:
+            fasta_files.append(input_file)
+        elif is_gff:
+            gff_files.append(input_file)
+        elif is_gbk:
+            gbk_files.append(input_file)
+        else:
+            raise click.BadParameter(
+                f"Unsupported file type: {input_file}. Supported: {SUPPORTED_FASTA_EXTS + SUPPORTED_GFF_EXTS + SUPPORTED_GBK_EXTS}"
+            )
+
+    # Read existing mapping if provided
+    old_to_new_mapping = None
+    if use_mapping:
+        old_to_new_mapping = read_mapping_file(use_mapping, from_col=from_col, to_col=to_col)
+
+    # Combined mapping from all FASTA files
+    combined_name_mapping = {}
+
+    # First pass: Rename all FASTA files
+    if fasta_files:
+        click.echo(f"Processing {len(fasta_files)} FASTA file(s)...")
+
+        for fasta_file in fasta_files:
+            input_path = Path(fasta_file)
+
+            if separate_by_size:
+                if len(fasta_files) > 1:
+                    raise click.BadParameter("--separate-by-size only supports a single FASTA input file")
+
+                output_prefix = str(outdir_path / input_path.stem)
+
+                click.echo(f"  Renaming FASTA with size separation: {fasta_file}")
```

Comment:
Is the space here on purpose for prettier printing?
```suggestion
                click.echo(f"Renaming FASTA with size separation: {fasta_file}")
```